src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5528,10 +5528,16 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_OutVariableCannotBeByRef" xml:space="preserve">
     <value>An out variable cannot be declared as a ref local</value>
   </data>
-  <data name="WRN_NullabilityMismatchInConstraints" xml:space="preserve">
+  <data name="WRN_NullabilityMismatchInConstraintsOnImplicitImplementation" xml:space="preserve">
     <value>Nullability in constraints for type parameter '{0}' of method '{1}' doesn't match the constraints for type parameter '{2}' of interface method '{3}'. Consider using an explicit interface implementation instead.</value>
   </data>
-  <data name="WRN_NullabilityMismatchInConstraints_Title" xml:space="preserve">
+  <data name="WRN_NullabilityMismatchInConstraintsOnImplicitImplementation_Title" xml:space="preserve">
     <value>Nullability in constraints for type parameter doesn't match the constraints for type parameter in implicitly implemented interface method'.</value>
   </data>
+  <data name="WRN_NullabilityMismatchInTypeParameterReferenceTypeConstraint" xml:space="preserve">
+    <value>The type '{2}' cannot be used as type parameter '{1}' in the generic type or method '{0}'. Nullability of type argument '{2}' doesn't match 'class' constraint.</value>
```

Comment:
>cannot [](start = 26, length = 6)

Consider using `should not` rather than `cannot` since this is a warning.

Reply:
> Consider using should not rather than cannot since this is a warning. 

Following an existing pattern established by WRN_NullabilityMismatchInTypeParameterConstraint for example

---
In reply to: [215055136](https://github.com/dotnet/roslyn/pull/29623#discussion_r215055136) [](ancestors = 215055136)

Reply:
Logged [https://github.com/dotnet/roslyn/issues/29697](https://github.com/dotnet/roslyn/issues/29697)

---
In reply to: [215363950](https://github.com/dotnet/roslyn/pull/29623#discussion_r215363950) [](ancestors = 215363950,215055136)